app/models/manageiq/providers/ibm_terraform/configuration_manager/configured_system.rb

```diff
@@ -1,8 +1,12 @@
 class ManageIQ::Providers::IbmTerraform::ConfigurationManager::ConfiguredSystem < ::ConfiguredSystem
   supports :console
-
   include ProviderObjectMixin
 
+  has_one :hardware, :through => :computer_system
+
+  virtual_delegate :cpu_total_cores, :to => :hardware, :allow_nil => true, :default => 0, :type => :integer
+  virtual_delegate :ram_size_mb,     :to => "hardware.memory_mb", :allow_nil => true, :default => 0, :type => :integer
```

Comment:
@k-chu if these are needed for chargeback we should probably add these to the base `ConfiguredSystem` model so chargeback doesn't blow up on e.g. foreman configured systems (even though they don't create hardwares they should have the methods)

Reply:
Thanks. Moved this to the base class.